lib/openstack_api/openstack_flavor.py

```diff
@@ -0,0 +1,127 @@
+        """
+        with self._connection_cls(cloud_account) as conn:
+            return conn.get_flavor(flavor_name)
+
+    def get_missing_flavors(self, source_cloud: str, dest_cloud: str) -> List[str]:
+        """
+        Get list of flavors missing in Dev cloud based on the flavors currently in
+        production
+        :param source_cloud: Cloud account for source cloud
+        :param dest_cloud: Cloud account for destination cloud
+        :return: List of names of missing flavors or empty list
+        """
+        # list flavors from prod
+        source_flavors = self.list_flavor(source_cloud)
+
+        # list flavors from dev
+        dest_flavors = self.list_flavor(dest_cloud)
+
+        # Prepare list of flavors that are in prod and dev
+        source_flavor_names = []
+        dest_flavor_names = []
+
+        for source in source_flavors:
+            source_flavor_names.append(source.name)
+
+        for dest in dest_flavors:
+            dest_flavor_names.append(dest.name)
+
+        # Find set of flavor names which are in prod but are missing from dev
+        missing_flavors = set(source_flavor_names).difference(set(dest_flavor_names))
+
+        return list(missing_flavors)
+
+    def create_flavor(self, cloud_account, flavor_data) -> Flavor:
+        """
+        Creates a new flavor
+        :param cloud_account: connection to cloud - should be dev cloud
+        :param flavor_data: Flavor object from production that we want to create in dev
+        :return: Flavor object or None if unsuccessful
+        """
+
+        if (
+            flavor_data.swap == ""
+        ):  # openstack stores swap=0 as swap='' when getting flavor information
+            flavor_swap = 0
+        else:
+            flavor_swap = flavor_data.swap
+
+        with self._connection_cls(cloud_account) as conn:
+            return conn.create_flavor(
+                name=flavor_data.name,
+                ram=flavor_data.ram,
+                vcpus=flavor_data.vcpus,
+                disk=flavor_data.disk,
+                flavorid="auto",
+                ephemeral=flavor_data.ephemeral,
+                swap=flavor_swap,
+                rxtx_factor=flavor_data.rxtx_factor,
+                is_public=flavor_data.is_public,
+            )
+
+    def get_flavor_specs(
+        self, cloud_account: str, flavor_id: Union[str, Flavor]
+    ) -> Dict:
+        """
+        Gets the metadata for a flavor
+        :param cloud_account: Cloud account to use
+        :param flavor_id: The id for a flavor or the Flavor object itself
+        :returns extra_specs: Dictionary containing flavor metadata
+        """
+
+        if isinstance(flavor_id, str):
+            flavor = self.get_flavor(cloud_account, flavor_id)
+            extra_specs = flavor.extra_specs
+        elif isinstance(flavor_id, Flavor):
+            extra_specs = flavor_id.extra_specs
+        else:
+            extra_specs = {}
```

Comment:
Could we change this to:

```
        if isinstance(flavor_id, str):
            flavor = self.get_flavor(cloud_account, flavor_id)
            extra_specs = flavor.extra_specs
        else:
            extra_specs = flavor_id.extra_specs
```

Since we should only get a flavor or string type here, so the `else {}` is a subtle bug waiting to happen